### PR TITLE
Fix a skipped test in inspector_service_test (was a false positive)

### LINKED ISSUE #6902: Re-enable a skipped test in inspector_service_test
A test has been excluded in PR #6903.

That test gives false positives for changes made in PR https://github.com/flutter/flutter/pull/138331. 

After that PR is merged, test fixtures must be updated and this test must be re-enabled.

affected files: [`inspector_service_test`](https://github.com/flutter/devtools/blob/7ff009753e57460fe3e82d0f22d4f4001b3cf996/packages/devtools_app/test/inspector/inspector_service_test.dart)

tests: [`widget tree`](https://github.com/flutter/devtools/blob/7ff009753e57460fe3e82d0f22d4f4001b3cf996/packages/devtools_app/test/inspector/inspector_service_test.dart#L383)

See also discussion on Discord ([link](https://discord.com/channels/608014603317936148/1179200000136319037/1181390182545641583))

### PR DESCRIPTION
**Pull request by bartekpacia**
Fixes #6902

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or there is a reason for not adding tests.

![build.yaml badge]

[build.yaml badge]: https://github.com/flutter/devtools/actions/workflows/build.yaml/badge.svg

### CODE REVIEW

**Comment by bartekpacia** [comment]
I'm not @goderbauer but since I added Semantics.identifier recently I'll say that it's not. I agree it can be removed when it's null.
If you agree I can make appropriate changes to the framework and to this PR.

**Comment by goderbauer** [comment]
> Fyi @goderbauer is the identifier property on Semantics important enough that we want to show it in the debugging output even when it is null?

@chunhtai Do you have an opinion here?

**Comment by chunhtai** [comment]
I don't think it should show up if it is null, the tooltip probably shouldn't show up as well. I can't see how they may be useful here.

@bartekpacia can you adjust the SemanticsProperties.debugFillProperties to give them a default value so it wouldn't show up here?

**Comment by bartekpacia** [comment]
I implemented @chunhtai's [suggestion from above](https://github.com/flutter/devtools/pull/6942#issuecomment-1852773200) in https://github.com/flutter/flutter/pull/140283, I think we now have to wait a few days until a framework roll into devtools.

**Comment by kenzieschmoll** [comment]
Will the golden output here need to be modified for your latest Flutter PR?

**Comment by bartekpacia** [comment]
Yes, after changes from my latest PR both `identifier` and `tooltip` won't be shown if they're `null`.

File: packages/devtools_app/test/test_infra/goldens/inspector_service_details_tree.txt [hunk 1]
```diff
@@ -12,6 +12,7 @@ MaterialApp
          └─Semantics
            │ container: false
            │ properties: SemanticsProperties
+           │ identifier: null
            │ tooltip: null
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/test_infra/goldens/inspector_service_details_tree.txt:16]: Don't these need to be removed now?

Comment (bartekpacia) [reply packages/devtools_app/test/test_infra/goldens/inspector_service_details_tree.txt]: That's right.

I was waiting for framework roll.

**Review by kenzieschmoll — APPROVED** [review]

**Comment by goderbauer** [comment]
Spoke to @kenzieschmoll about this last week and she said she's gonna take another look at this.

**Comment by bartekpacia** [comment]
Thanks for the update @goderbauer!

**Pull request merged**